src/qibolab/instruments/rfsoc.py

```diff
@@ -93,39 +92,32 @@ def convert_sweep(sweeper: Sweeper, sequence: PulseSequence, qubits: List[Qubit]
     if sweeper.parameter is Parameter.bias:
         for qubit in sweeper.qubits:
             parameters.append(rfsoc.Parameter.bias)
-            for idx, seq_qubit in enumerate(qubits):
-                if qubit == seq_qubit:
-                    indexes.append(idx)
-            starts.append(sweeper.values[0] + qubits[qubit].flux.bias)
-            stops.append(sweeper.values[-1] + qubits[qubit].flux.bias)
+            indexes.append(list(qubits.values().index(qubit)))
+
+            starts.append(sweeper.values[0] + qubit.flux.bias)
+            stops.append(sweeper.values[-1] + qubit.flux.bias)
 
         if max(np.abs(starts)) > 1 or max(np.abs(stops)) > 1:
             raise ValueError("Sweeper amplitude is set to reach values higher than 1")
     else:
         for pulse in sweeper.pulses:
-            for idx, seq_pulse in enumerate(sequence):
-                if pulse == seq_pulse:
-                    indexes.append(idx)
-
-            if sweeper.parameter is Parameter.amplitude:
-                parameters.append(rfsoc.Parameter.amplitude)
-                starts.append(sweeper.values[0] * pulse.amplitude)
-                stops.append(sweeper.values[-1] * pulse.amplitude)
-            elif sweeper.parameter is Parameter.relative_phase:
-                parameters.append(rfsoc.Parameter.relative_phase)
-                starts.append(np.degrees(sweeper.values[0] + pulse.relative_phase))
-                stops.append(np.degrees(sweeper.values[-1] + pulse.relative_phase))
-            elif sweeper.parameter is Parameter.frequency:
-                # frequency are not convertet yet and LO is not accounted for
-                parameters.append(rfsoc.Parameter.frequency)
-                starts.append(sweeper.values[0] + pulse.frequency)
-                stops.append(sweeper.values[-1] + pulse.frequency)
+            indexes.append(sequence.index(pulse))
+
+            name = sweeper.parameter.name
+            parameters.append(getattr(rfsoc.Parameter, name))
+            value = getattr(pulse, name)
+            if sweeper.parameter in {Parameter.frequency, Parameter.relative_phase}:
+                starts.append(sweeper.values[0] + value)
+            elif sweeper.parameter is Parameter.amplitude:
+                starts.append(sweeper.values[0] * value)
```

Comment:
Why didn't you join even these two? And where are the `stops` now?

Reply:
I just forgot the stops. The two options are different because the first values gets multiplied and the second summed

Reply:
Ah, you're right, I haven't noticed.

In principle, we could join even in this case, but let's keep it simpler, at least for the time being.